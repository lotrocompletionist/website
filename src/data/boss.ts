export interface IBoss {
  id: number;
  name: string;
}
