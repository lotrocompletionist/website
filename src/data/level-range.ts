export interface ILevelRange {
  minimum: number;
  maximum?: number;
}
