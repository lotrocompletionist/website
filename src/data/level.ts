export interface Level {
  minimum: number;
  maximum?: number;
}
