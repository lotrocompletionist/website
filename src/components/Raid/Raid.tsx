import * as React from "react";
import {
  CardContent,
  Media,
  MediaLeft,
  MediaContent,
  Content,
  Image,
  Title,
  Subtitle
} from "bloomer";
import { Level } from "../Level/Level";
import { Bosses } from "../Bosses/Bosses";
import { IRaid } from "../../data/raid";

export interface RaidProps {
  raid: IRaid;
}

export function Raid(props: RaidProps) {
  return (
    <>
      <CardContent>
        <Media>
          <MediaLeft>
            <Image isSize="48x48" src="https://via.placeholder.com/96x96" />
          </MediaLeft>
          <MediaContent>
            <Title isSize={4}>{props.raid.name}</Title>
            <Subtitle isSize={6}>
              {props.raid.abbreviations.join(", ")}
            </Subtitle>
          </MediaContent>
        </Media>
        <Content>
          {props.raid.cluster}
          <br />
          <small>
            <Level level={props.raid.level} />
            <br />
            <Bosses bosses={props.raid.bosses} />
          </small>
        </Content>
      </CardContent>
    </>
  );
}
