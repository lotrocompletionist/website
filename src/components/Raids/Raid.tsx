import * as React from "react";
import {
  CardContent,
  Media,
  MediaLeft,
  MediaContent,
  Content,
  Image,
  Title,
  Subtitle
} from "bloomer";
import { LevelRange } from "@app/components/Shared/LevelRange/LevelRange";
import { Bosses } from "@app/components/Shared/Bosses/Bosses";
import { IRaid } from "@app/data/raid";

export interface RaidProps {
  raid: IRaid;
}

export function Raid(props: RaidProps) {
  return (
    <>
      <CardContent>
        <Media>
          <MediaLeft>
            <Image isSize="48x48" src="https://via.placeholder.com/96x96" />
          </MediaLeft>
          <MediaContent>
            <Title isSize={4}>{props.raid.name}</Title>
            <Subtitle isSize={6}>
              {props.raid.abbreviations.join(", ")}
            </Subtitle>
          </MediaContent>
        </Media>
        <Content>
          {props.raid.cluster}
          <br />
          <small>
            <LevelRange range={props.raid.level} />
            <br />
            <Bosses bosses={props.raid.bosses} />
          </small>
        </Content>
      </CardContent>
    </>
  );
}
