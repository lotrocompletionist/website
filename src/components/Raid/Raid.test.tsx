import React from "react";
import renderer from "react-test-renderer";
import { Raid } from "./Raid";
import { Challenge, Availability, Tiers, Group } from "../../data/models";

test("<Raid> renders correctly", () => {
  const raid = {

    "name": "Helegrod: Spider Wing",
    "abbreviations": [
        "Spider-wing"
    ],
    "region": "The Misty Mountains",
    "level": {
        "minimum": 50,
        "maximum": 115
    },
    "bossEncounters": 1,
    "tiers": Tiers.One,
    "groups": [
        Group.Raid
    ],
    "availability": [
        Availability.VIP,
        Availability.QuestPack
    ],
    "prerequisites": "",
    "challenge": Challenge.Daily,
    "cluster": "Shadows of Angmar"
};
  const tree = renderer.create(<Raid raid={raid} />).toJSON();
  expect(tree).toMatchSnapshot();
});
