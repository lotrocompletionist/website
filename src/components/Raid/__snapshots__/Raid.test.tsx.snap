// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Raid> renders correctly 1`] = `
<div
  className="card-content"
>
  <article
    className="media"
  >
    <div
      className="media-left"
    >
      <figure
        className="image is-48x48"
      >
        <img
          src="https://via.placeholder.com/96x96"
        />
      </figure>
    </div>
    <div
      className="media-content"
    >
      <h1
        className="title is-4"
      >
        Helegrod: Drake Wing
      </h1>
      <h2
        className="subtitle is-6"
      >
        Drake-wing
      </h2>
    </div>
  </article>
  <div
    className="content"
  >
    Shadows of Angmar
    <br />
    <small>
      Level: 
      50
       - 115
      <br />
      Bosses: 
      Grisgart, Praunk-Goj, RÃ©mgoj, Burzdyr, Burzgost
    </small>
  </div>
</div>
`;
